### PR TITLE
Updated README to include a bit more on getting started writing a plugin

### PR DESCRIPTION
**Pull request by apetrynet**
As mentioned in the latest TSC meeting, we were lacking a bit of documentation about getting started using this template repo and how to write plugins.
This PR attempts to fill in a few gaps and also includes links to relevant documentation.
 
Signed-off-by: apetrynet <flehnerheener@gmail.com>

### CODE REVIEW

**Comment by apetrynet** [comment]
@ssteinbach, @reinecke 
I think we should land [this](https://github.com/AcademySoftwareFoundation/OpenTimelineIO/pull/1378) PR before we land this one so the links to the documentation don't confuse people on what we want to do with contrib adapters

**Comment by apetrynet** [comment]
Should we mention something about contrib adapters moving out of the main repo? Or is it sufficient with what's mentioned in @reinecke's PR?

File: README.md [hunk 1]
```diff
@@ -44,40 +65,49 @@ Examples:
...
+  
+You're free to rename, remove or restructure the files and folders below, 
+but make sure the `plugin_manifest.json` file is kept and that the contents 
+inside it reflect your choices so OpenTimelineIO's plugin system loads your 
+plugin properly.
+
+Make sure to add a descriptive docstring at the top of your plugin files, so they 
+register properly and inform users of what they do.
+
+```commandline
```
Comment (JeanChristopheMorinPerso) [inline README.md:78]:
```suggestion
```
```
`commandline` is not necessary here.

Comment (apetrynet) [reply README.md:78]: Thanks!
I'll remove those from other blocks as well.

File: README.md [hunk 2]
```diff
@@ -43,41 +64,81 @@ Examples:
...
+inside it reflect your choices. This makes sure OpenTimelineIO's plugin system 
+loads your plugin properly.
+
+> **TIP!** Make sure to add a descriptive docstring at the top of your plugin files, so they 
+register properly and inform users of what they do.
+
+Example of a simple adapter plugin. Notice how we removed the "adapters" folder.
+```
+ |── LICENSE
+ ├── otio_my_adapter
```
Comment (JeanChristopheMorinPerso) [inline README.md:107]: I would still recommend using the `src` layout. Using a `src` folder can avoid some errors while running tests (for example it ensures that tests are not ran against the non installed code (unless you mess with your PYTHONPATH).

Basically, when using the `src` layout, `otio_my_adapter` is not importable when python is invoked from the root of the repo. WHich means that to run the tests in CI, you'll have to actually install the project using `pip install .`, which will test the code and also test that the packaging works as expected (al files are packaged in the wheel, etc).

Comment (apetrynet) [reply README.md:107]: Thanks!
This makes a lot of sense!
I think I'll do the restructuring of code in a separate PR though.

File: README.md [hunk 3]
```diff
@@ -43,41 +64,81 @@ Examples:
...
+            "filepath" : "my_adapter.py",
+            "suffixes" : ["xyz"]
+        }
+    ]
+}
+```
 
 ## Testing your plugin during development
-```bash
+```
```
Comment (reinecke) [inline README.md:135]: This is just a general question for my own information, why did you remove the syntax highlighting hint? Was it causing issues? I only ask because I use them all the time and I'm just curious if maybe I shouldn't :)

**Review by reinecke — APPROVED** [review]
Nice updates!

**Comment by apetrynet** [comment]
Even though the block looked good without any hinting, I initially added
"bash" to make s it clear I was in a commandline. I later switched to the
"commandline" hint based on suggestions from pycharm and the fact that not
all OS use bash.
After some feedback I removed the hinting all together as the code block
seems to come through fine without it.

I have no issues adding them again if it makes more sense.

**Review by ssteinbach — APPROVED** [review]
LGTM! Nice work, @apetrynet

**Comment by apetrynet** [comment]
Great!
Thanks @ssteinbach .
Should we land this now that @reinecke's PR is live?

**Comment by ssteinbach** [comment]
Go for it!  I wonder if we can work "community updates" or something into the release email we send out when we release the main OTIO library so we can coordinate some of this stuff a bit.  Even just a note "updated plugin template".

**Comment by apetrynet** [comment]
Nice! I don't have the proper permissions to merge. 
Agree about noting relevant updates from the OTIO organisation's repositories at release time.
